src/main/java/net/coderbot/iris/pipeline/TransformPatcher.java

```diff
@@ -0,0 +1,512 @@
+		 * PREV TODO: Only do the NewLines patches if the source code isn't from
+		 * gbuffers_lines
+		 */
+
+		TokenFilter<Parameters> parseTokenFilter = new ChannelFilter<Parameters>(TokenChannel.PREPROCESSOR) {
+			@Override
+			public boolean isTokenAllowed(Token token) {
+				if (!super.isTokenAllowed(token)) {
+					throw new SemanticException("Unparsed preprocessor directives such as '" + token.getText()
+							+ "' may not be present at this stage of shader processing!");
+				}
+				return true;
+			}
+		};
+
+		// setup the transformations and even loose phases if necessary
+		LifecycleUser<Parameters> detectReserved = new SearchTerminals<Parameters>()
+				.singleTarget(
+						new ThrowTargetImpl<Parameters>(
+								"iris_",
+								"Detected a potential reference to unstable and internal Iris shader interfaces (iris_). This isn't currently supported."))
+				.requireFullMatch(false);
+
+		// #region patchAttributes
+		LifecycleUser<Parameters> replaceEntityColorDeclaration = new Transformation<Parameters>() {
+			@Override
+			protected void setupGraph() {
+				addEndDependent(new WalkPhase<Parameters>() {
+					ParseTreePattern entityColorPattern;
+
+					@Override
+					public void init() {
+						entityColorPattern = compilePattern(
+								"uniform vec4 entityColor;",
+								GLSLParser.RULE_externalDeclaration);
+					}
+
+					@Override
+					public void enterExternalDeclaration(ExternalDeclarationContext ctx) {
+						ParseTreeMatch match = entityColorPattern.match(ctx);
+						if (match.succeeded()) {
+							removeNode(ctx);
+						}
+					}
+				});
+
+				if (getJobParameters().type == ShaderType.GEOMETRY) {
+					chainDependent(new SearchTerminals<Parameters>().singleTarget(
+							new ParsedReplaceTargetImpl<>("entityColor", "entityColor[0]", GLSLParser::expression)));
+				}
+
+				chainDependent(new RunPhase<Parameters>() {
+					@Override
+					protected void run(TranslationUnitContext ctx) {
+						switch (getJobParameters().type) {
+							case VERTEX:
+								injectExternalDeclarations(InjectionPoint.BEFORE_DECLARATIONS,
+										"uniform sampler2D iris_overlay;",
+										"varying vec4 entityColor;");
+								break;
+							case GEOMETRY:
+								injectExternalDeclarations(InjectionPoint.BEFORE_DECLARATIONS,
+										"out vec4 entityColorGS;",
+										"in vec4 entityColor[];");
+								break;
+							case FRAGMENT:
+								injectExternalDeclaration(InjectionPoint.BEFORE_DECLARATIONS, "varying vec4 entityColor;");
+								break;
+						}
+					}
+				});
+			}
+		};
+
+		LifecycleUser<Parameters> wrapOverlay = new MainWrapper<Parameters>() {
+			@Override
+			protected String getMainContent() {
+				return getJobParameters().type == ShaderType.VERTEX
+						? "vec4 overlayColor = texture2D(iris_overlay, (gl_TextureMatrix[2] * gl_MultiTexCoord2).xy);\n" +
+								"entityColor = vec4(overlayColor.rgb, 1.0 - overlayColor.a);\nirisMain();"
+						: "entityColorGS = entityColor[0];\nirisMain();";
```

Comment:
This doesn't seem quite right

Reply:
I'll have to look into how this is in the original